package/run.sh

```diff
@@ -2,9 +2,27 @@
 
 set -x -e
 
-cp -f ${CATTLE_AGENT_EXECUTION_PWD}/k3s /usr/local/bin/k3s
-chmod 755 /usr/local/bin/k3s
-chown root:root /usr/local/bin/k3s
+SA_INSTALL_PREFIX="/usr/local"
```

Comment:
There's an additional part to this that's missing - the prefix is passed through to the RKE2 installer as `INSTALL_RKE2_TAR_PREFIX` :
https://github.com/rancher/system-agent-installer-rke2/blob/708ad277b89bd7e68fcd0bd959c7df3ffc804b74/package/run.sh#L52

K3s has an equivalent `INSTALL_K3S_BIN_DIR` variable that is already set to /opt/bin if /usr/local/bin is read-only:
https://github.com/k3s-io/k3s/blob/6c544a46796e7dc53d73fdd75f498ee37c3e8e95/install.sh#L227-L233

So I'm honestly not sure that this is needed?